c/tskit/trees.c

```diff
@@ -3755,8 +3744,12 @@ tsk_treeseq_allele_frequency_spectrum(const tsk_treeseq_t *self,
         num_time_windows = 1;
         time_windows = default_time_windows;
     } else {
-        ret = tsk_treeseq_check_time_windows(
-            num_time_windows, time_windows);
+        if (stat_site
+            && tsk_memcmp(time_windows, default_time_windows, sizeof(double)) != 0) {
```

Comment:
Hm, this is a bit awkward - what if instead we used `num_time_windows=0` to mean "default/no time windows"?

Reply:
Oh, wait - we're already *in* the `else` clause where we know that `time_windows != NULL`. So I don't think we need to check this at all - just throw the error?
```suggestion
            ) {
```

Reply:
So, if someone explicitly specifies `time_windows = [0, np.inf], mode="node"` then they'll get the error, but that's okay - the error message says "you can't specify time windows", not "time windows must be [0, Inf)" (I think that's what it says anyhow).